### ISSUE TITLE
XMLHttpRequest

### ISSUE THREAD

**Issue by sima-qian** [fac-14/NDJS-API#27]
Is there a reason for this line:
```
if (xhr.status >= 200 && xhr.status < 300)
```
rather than
```
if (xhr.readyState == 4 && xhr.status == 200)
```
as used in workshops this week? You accept any status code that starts with a 2 -- what's the benefit of this?

**Comment by VirtualDOMinic** [comment]
@sima-qian 

Status code:
Not sure they way we've done it is necessarily better, but back when I did my first XHR a few months back, I read an example which checked for any response between 200 and 299, which makes sense to me since any 2XX response code is a "success". https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#2xx_Success - looking at this, I'm not sure we'd ever expect a 2XX response that isn't 200 though!

readyState:
Instead of readyState == 4, we've used the eventlistener to listen for the ready state. This made more sense to me, because in my mind the eventlistener is always waiting for the success, while an "if" statement wouldn't check again if the readyState weren't 4 on the first check...

Does this sort of make sense?
